Index: Trunk/src/map/battle.c
===================================================================
--- Trunk/src/map/battle.c	(revision 263)
+++ Trunk/src/map/battle.c	(working copy)
@@ -1448,7 +1448,7 @@
 				break;
 
 			case CR_SHIELDBOOMERANG:
-			case PA_SHIELDCHAIN:
+			//case PA_SHIELDCHAIN:
 			case LG_SHIELDPRESS:
 			case LG_EARTHDRIVE:
 				flag.weapon = 0;
@@ -1798,7 +1798,7 @@
 				}
 				break;
 			case CR_SHIELDBOOMERANG:
-			case PA_SHIELDCHAIN:
+			//case PA_SHIELDCHAIN:
 			case LG_SHIELDPRESS:
 			case LG_EARTHDRIVE:
 				wd.damage = sstatus->batk;
@@ -1822,7 +1822,7 @@
 					if (index >= 0 &&
 						sd->inventory_data[index] &&
 						sd->inventory_data[index]->type == IT_WEAPON)
-						wd.damage = sd->inventory_data[index]->weight/5;
+						wd.damage = sd->inventory_data[index]->weight;
 				} else
 					wd.damage = sstatus->rhw.atk2*2;
 				i=100+(40*(skill_lv-1));
@@ -2324,8 +2324,16 @@
 		{ //Refine bonus applies after cards and elements.
 			short index= sd->equip_index[EQI_HAND_L];
 			if( index >= 0 && sd->inventory_data[index] && sd->inventory_data[index]->type == IT_ARMOR )
-				ATK_ADD(10*sd->status.inventory[index].refine);
+				ATK_ADD(25*sd->status.inventory[index].refine);
 		}
+		if ( skill_num == PA_SHIELDCHAIN ) {
+			short index = sd->equip_index[EQI_HAND_L];
+			if (index >= 0 && sd->inventory_data[index] && sd->inventory_data[index]->type == IT_ARMOR) {
+				ATK_ADD(sd->inventory_data[index]->weight*3);
+			}
+			if( sc && sc->data[SC_GLOOMYDAY_SK] )
+				ATK_ADD(50 + 5 * sc->data[SC_GLOOMYDAY_SK]->val1);
+		}
 	} //if (sd)
 
 	//Card Fix, tsd sid
@@ -2633,7 +2641,7 @@
 			break;
 
 		case CR_SHIELDBOOMERANG:
-		case PA_SHIELDCHAIN:
+		//case PA_SHIELDCHAIN:
 			flag.weapon = 0;
 			break;
 
@@ -2960,7 +2968,7 @@
 			wd.damage2 = 0;
 			break;
 		case CR_SHIELDBOOMERANG:
-		case PA_SHIELDCHAIN:
+		//case PA_SHIELDCHAIN:
 		case LG_SHIELDPRESS:
 		case LG_EARTHDRIVE:
 			wd.damage = sstatus->batk;
@@ -3302,7 +3310,7 @@
 		{
 		case NC_AXEBOOMERANG:
 			if( sd && (i = sd->equip_index[EQI_HAND_R]) >= 0 && sd->inventory_data[i] && sd->inventory_data[i]->type == IT_WEAPON )
-				i = sd->inventory_data[i]->weight;
+				i = (sd->inventory_data[i]->weight*3);
 			break;
 		case MO_EXTREMITYFIST:
 			i = 250 + 150 * skill_lv;
@@ -3484,8 +3492,17 @@
 	BON_RATE2(c_bossmod[0],c_bossmod[1]);
 	BON_RATE(c_atkmod[0]);
 	if( sd && (skill_id == CR_SHIELDBOOMERANG || skill_id == PA_SHIELDCHAIN) && (i = sd->equip_index[EQI_HAND_L]) >= 0 && sd->inventory_data[i] && sd->inventory_data[i]->type == IT_ARMOR )
-		BON_ADD(10 * sd->status.inventory[i].refine);
+		BON_ADD(25 * sd->status.inventory[i].refine);
 
+	if ( skill_id == PA_SHIELDCHAIN ) {
+		short index = sd->equip_index[EQI_HAND_L];
+		if (index >= 0 && sd->inventory_data[index] && sd->inventory_data[index]->type == IT_ARMOR) {
+			ATK_ADD(sd->inventory_data[index]->weight*3);
+		}
+		if( sc && sc->data[SC_GLOOMYDAY_SK] )
+			ATK_ADD(50 + 5 * sc->data[SC_GLOOMYDAY_SK]->val1);
+	}
+
 	if( flag.hit && !flag.infdef )
 	{
 		// -------------------------------------------------
@@ -3676,13 +3693,24 @@
 	switch( skill_id )
 	{
 	case CR_ACIDDEMONSTRATION:
+		{
+			struct Damage md = battle_calc_magic_attack_renewal(src,target,skill_id,skill_lv,wflag);
+			wd.damage += md.damage;
+			if( skill_id == CR_ACIDDEMONSTRATION && target->type == BL_PC ){
+				wd.damage = wd.damage/(90/10);
+				wd.damage >>= 1; // Half Damage on Players
+			}
+			else
+			{
+				wd.damage = wd.damage/(77/10);
+			}
+		}
+		break;
 	case ASC_BREAKER:
 	case LG_RAYOFGENESIS:
 		{
 			struct Damage md = battle_calc_magic_attack_renewal(src,target,skill_id,skill_lv,wflag);
 			wd.damage += md.damage;
-			if( skill_id == CR_ACIDDEMONSTRATION && target->type == BL_PC )
-				wd.damage >>= 1; // Half Damage on Players
 		}
 		break;
 	case 0: // Melee Attacks
Index: Trunk/src/map/skill.c
===================================================================
--- Trunk/src/map/skill.c	(revision 263)
+++ Trunk/src/map/skill.c	(working copy)
@@ -852,7 +852,7 @@
 		skillratio += 10 * skill_lv - 10;
 		break;
 	case PA_SHIELDCHAIN:
-		skillratio += 30 * skill_lv;
+		skillratio += 100 + 30 * skill_lv;
 		break;
 	case WS_CARTTERMINATION:
 		i = 10 * (16 - skill_lv);
@@ -3270,7 +3270,7 @@
 
 	damage = dmg.damage + dmg.damage2;
 
-	if( (skillid == AL_INCAGI || skillid == AL_BLESSING || skillid == CASH_BLESSING || skillid == CASH_INCAGI || skillid == MER_INCAGI || skillid == MER_BLESSING) && tsd->sc.data[SC_CHANGEUNDEAD] )
+	if( (skillid == AL_INCAGI || skillid == AL_BLESSING || skillid == CASH_BLESSING || skillid == CASH_INCAGI || skillid == MER_INCAGI || skillid == MER_BLESSING || skillid == AB_CANTO || skillid == AB_CLEMENTIA) && tsd->sc.data[SC_CHANGEUNDEAD] )
 		damage = 1;
 
 	if( damage > 0 && ((dmg.flag&BF_WEAPON && src != bl && ( src == dsrc || ( dsrc->type == BL_SKILL && ( skillid == SG_SUN_WARM || skillid == SG_MOON_WARM || skillid == SG_STAR_WARM ) ) )
@@ -6314,11 +6314,15 @@
 	case AL_RUWACH:
 	case WZ_SIGHTBLASTER:
 	case NPC_WIDESIGHT:
-	case NPC_STONESKIN:
 	case NPC_ANTIMAGIC:
 		clif_skill_nodamage(src,bl,skillid,skilllv,
 			sc_start2(bl,type,100,skilllv,skillid,skill_get_time(skillid,skilllv)));
 		break;
+	case NPC_STONESKIN:
+		clif_specialeffect(src, 214, AREA);
+		clif_skill_nodamage(src,bl,skillid,skilllv,
+			sc_start2(bl,type,100,skilllv,skillid,skill_get_time(skillid,skilllv)));
+		break;
 	case HLIF_AVOID:
 	case HAMI_DEFENCE:
 		i = skill_get_time(skillid,skilllv);
@@ -6942,6 +6946,7 @@
 			status_change_end(bl, SC_STONE, INVALID_TIMER);
 			status_change_end(bl, SC_SLEEP, INVALID_TIMER);
 			status_change_end(bl, SC_STUN, INVALID_TIMER);
+			status_change_end(bl, SC_WHITEIMPRISON, INVALID_TIMER);
 		}
 		//Is this equation really right? It looks so... special.
 		if(battle_check_undead(tstatus->race,tstatus->def_ele))
