Index: src-base/map/battle.c
===================================================================
--- src-base/map/battle.c	(revision 238)
+++ src-base/map/battle.c	(working copy)
@@ -1448,7 +1448,7 @@
 				break;
 
 			case CR_SHIELDBOOMERANG:
-			case PA_SHIELDCHAIN:
+			//case PA_SHIELDCHAIN:
 			case LG_SHIELDPRESS:
 			case LG_EARTHDRIVE:
 				flag.weapon = 0;
@@ -1798,7 +1798,7 @@
 				}
 				break;
 			case CR_SHIELDBOOMERANG:
-			case PA_SHIELDCHAIN:
+			//case PA_SHIELDCHAIN:
 			case LG_SHIELDPRESS:
 			case LG_EARTHDRIVE:
 				wd.damage = sstatus->batk;
@@ -2323,9 +2323,19 @@
 		if( skill_num == CR_SHIELDBOOMERANG || skill_num == PA_SHIELDCHAIN )
 		{ //Refine bonus applies after cards and elements.
 			short index= sd->equip_index[EQI_HAND_L];
-			if( index >= 0 && sd->inventory_data[index] && sd->inventory_data[index]->type == IT_ARMOR )
+			if( index >= 0 && sd->inventory_data[index] && sd->inventory_data[index]->type == IT_ARMOR ) {
 				ATK_ADD(10*sd->status.inventory[index].refine);
+			}
 		}
+		if ( skill_num == PA_SHIELDCHAIN ) {
+			short index = sd->equip_index[EQI_HAND_L];
+			if (index >= 0 && sd->inventory_data[index] && sd->inventory_data[index]->type == IT_ARMOR) {
+				ATK_ADD(sd->inventory_data[index]->weight/10);
+			}
+			if( sc && sc->data[SC_GLOOMYDAY_SK] )
+				ATK_ADD(50 + 5 * sc->data[SC_GLOOMYDAY_SK]->val1);
+		}
+		
 	} //if (sd)
 
 	//Card Fix, tsd sid
@@ -2633,7 +2643,7 @@
 			break;
 
 		case CR_SHIELDBOOMERANG:
-		case PA_SHIELDCHAIN:
+		//case PA_SHIELDCHAIN:
 			flag.weapon = 0;
 			break;
 
@@ -2960,7 +2970,7 @@
 			wd.damage2 = 0;
 			break;
 		case CR_SHIELDBOOMERANG:
-		case PA_SHIELDCHAIN:
+		//case PA_SHIELDCHAIN:
 		case LG_SHIELDPRESS:
 		case LG_EARTHDRIVE:
 			wd.damage = sstatus->batk;
@@ -3478,14 +3488,24 @@
 	ATK_RATE2(c_bossmod[0],c_bossmod[1]); // Add Boss Modifications
 	ATK_RATE(c_atkmod[0]);
 	ATK_ADD(base_damage); // Adding Status Damage to Formula.
-
+	
 	// This cannot be added until def/def2 reductions are done
 	BON_RATE2(c_racemod[0],c_racemod[1]);
 	BON_RATE2(c_bossmod[0],c_bossmod[1]);
 	BON_RATE(c_atkmod[0]);
-	if( sd && (skill_id == CR_SHIELDBOOMERANG || skill_id == PA_SHIELDCHAIN) && (i = sd->equip_index[EQI_HAND_L]) >= 0 && sd->inventory_data[i] && sd->inventory_data[i]->type == IT_ARMOR )
+	if( sd && (skill_id == CR_SHIELDBOOMERANG || skill_id == PA_SHIELDCHAIN) && (i = sd->equip_index[EQI_HAND_L]) >= 0 && sd->inventory_data[i] && sd->inventory_data[i]->type == IT_ARMOR ) {
 		BON_ADD(10 * sd->status.inventory[i].refine);
+	}
 
+	if ( skill_id == PA_SHIELDCHAIN ) {
+		short index = sd->equip_index[EQI_HAND_L];
+		if (index >= 0 && sd->inventory_data[index] && sd->inventory_data[index]->type == IT_ARMOR) {
+			ATK_ADD(sd->inventory_data[index]->weight/10);
+		}
+		if( sc && sc->data[SC_GLOOMYDAY_SK] )
+			ATK_ADD(50 + 5 * sc->data[SC_GLOOMYDAY_SK]->val1);
+	}
+
 	if( flag.hit && !flag.infdef )
 	{
 		// -------------------------------------------------
Index: src-base/map/skill.c
===================================================================
--- src-base/map/skill.c	(revision 238)
+++ src-base/map/skill.c	(working copy)
@@ -852,7 +852,7 @@
 		skillratio += 10 * skill_lv - 10;
 		break;
 	case PA_SHIELDCHAIN:
-		skillratio += 30 * skill_lv;
+		skillratio += 100 + 30 * skill_lv;
 		break;
 	case WS_CARTTERMINATION:
 		i = 10 * (16 - skill_lv);
@@ -6937,6 +6937,7 @@
 			status_change_end(bl, SC_STONE, INVALID_TIMER);
 			status_change_end(bl, SC_SLEEP, INVALID_TIMER);
 			status_change_end(bl, SC_STUN, INVALID_TIMER);
+			status_change_end(bl, SC_WHITEIMPRISON, INVALID_TIMER);
 		}
 		//Is this equation really right? It looks so... special.
 		if(battle_check_undead(tstatus->race,tstatus->def_ele))
Index: src-base/map/status.c
===================================================================
--- src-base/map/status.c	(revision 238)
+++ src-base/map/status.c	(working copy)
@@ -7071,23 +7071,17 @@
 			val2 = 75 + 25*val1; //Cri bonus
 			break;
 		case SC_ASPDPOTION0:
+			val2 = 40;
+			break;
 		case SC_ASPDPOTION1:
+			val2 = 60;
+			break;
 		case SC_ASPDPOTION2:
+			val2 = 90;
+			break;
 		case SC_ASPDPOTION3:
-			if( battle_config.renewal_system_enable )
-			{
-				switch( type )
-				{
-				case SC_ASPDPOTION0: val2 = 40; break;
-				case SC_ASPDPOTION1: val2 = 60; break;
-				case SC_ASPDPOTION2: val2 = 90; break;
-				case SC_ASPDPOTION3: val2 = 120; break; // Where is this used?
-				}
-			}
-			else
-				val2 = 50*(2+type-SC_ASPDPOTION0);
+			val2 = 120;
 			break;
-
 		case SC_WEDDING:
 		case SC_XMAS:
 		case SC_SUMMER:
